web/react/components/user_settings.jsx

```diff
@@ -624,6 +624,16 @@ var SecurityTab = React.createClass({
     }
 });
 
+var checkServerError = function(error) {
```

Comment:
The determination that we have a duplicate is done internal to the sql query, so we have to intercept the message and convert it somewhere. I chose to do it client side because we have a history of converting error messages at this layer.